cassandra/lib/JSONSchema.js

```diff
@@ -225,6 +208,60 @@ class JSONSchema {
         return this;
     }
 
+    /**
+     * Returns true if schema contains same columns as metadata
+     * @param metadataDescriptor cassandra-driver metadata object
+     * @returns {boolean}
+     */
+    compareColumnsSetWithMetadata(metadataDescriptor) {
+        return Metadata.create(metadataDescriptor).isSameMembersSchema(this);
+    }
+
+    /**
+     * Returns array of column types mismatches in schema with metadata
+     * @param metadataDescriptor cassandra-driver metadata object
+     * @returns {Array}
+     */
+    diffColumnTypesWithMetadata(metadataDescriptor) {
+        const mismatches = [];
+        const metadata = Metadata.create(metadataDescriptor);
+
+        const columns = this.getColumns();
+
+        for (let colName in columns) {
+            if (metadata.columnExists(colName)) {
+                const realType = metadata.getColumnFullTypeName(colName);
```

Comment:
Please move body of this condition to separate method in Metadata, i.e. isTypeMismatch(colName) or just mismatch(colName). Also rename colName and everything related to columns to more generic names, like memberName, members etc. because this is used not only for columns but for fields of UDT as well